lib/assets/chart_builder/main.js

```diff
@@ -0,0 +1,223 @@
+            </div>
+            <div class="field">
+              <label class="input-label">Height</label>
+              <input class="input input--xs" name="height" type="number" />
+            </div>
+          </div>
+        </div>
+        <div class="row">
+          <div class="group">
+            <div class="field">
+              <label class="input-label">X Axis</label>
+              <select class="input input--sm" name="x_axis">
+              </select>
+            </div>
+            <div class="field">
+              <label class="input-label">Type</label>
+              <select class="input input--sm" name="x_axis_type">
+                <option value=""></option>
+                <option value="quantitative">Quantitative</option>
+                <option value="nominal">Nominal</option>
+                <option value="ordinal">Ordinal</option>
+              </select>
+            </div>
+          </div>
+          <div class="group">
+            <div class="field">
+              <label class="input-label">Y Axis</label>
+              <select class="input input--sm" name="y_axis">
+              </select>
+            </div>
+            <div class="field">
+              <label class="input-label">Type</label>
+              <select class="input input--sm" name="y_axis_type">
+                <option value=""></option>
+                <option value="quantitative">Quantitative</option>
+                <option value="nominal">Nominal</option>
+                <option value="ordinal">Ordinal</option>
+              </select>
+            </div>
+          </div>
+          <div class="group">
+            <div class="field">
+              <label class="input-label">Color</label>
+              <select class="input input--sm" name="color">
+              </select>
+            </div>
+            <div class="field">
+              <label class="input-label">Type</label>
+              <select class="input input--sm" name="color_type">
+                <option value=""></option>
+                <option value="quantitative">Quantitative</option>
+                <option value="nominal">Nominal</option>
+                <option value="ordinal">Ordinal</option>
+              </select>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `;
+
+  updateInfoBox(info.missing_dep);
+  setDataOptions(info.options);
+  setValues(info.fields);
+
+  const hasType = ["x_axis", "y_axis", "color"];
+  const types = ["x_axis_type", "y_axis_type", "color_type"];
+
+  ctx.root.addEventListener("blur", handleFieldChange, true);
+  ctx.root.addEventListener("change", handleFieldChange);
+
+  function handleFieldChange(event) {
+    const { name, value } = event.target;
+    if (hasType.includes(name)) { maybeReset({name, value}) }
+    ctx.pushEvent("update_field", { field: name, value });
+  }
+
+  ctx.handleEvent("update", ({ fields }) => {
+    setValues(fields);
+  });
```

Comment:
```suggestion
  ctx.handleEvent("update", ({ fields }) => setValues(fields));
```

can't these be shortened to one liners like so?